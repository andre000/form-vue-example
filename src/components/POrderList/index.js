export { default as POrderList } from './POrderList.vue'
