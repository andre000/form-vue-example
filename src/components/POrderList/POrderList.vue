<template>
  <ul class="order-list">
    <p-order-item v-for="(order, i) in orders" :key="i" :order="order" />
  </ul>
</template>

<script>
import { POrderItem } from '@/components/POrderItem'

export default {
  name: 'POrderList',
  components: {
    POrderItem
  },

  props: {
    orders: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style></style>
