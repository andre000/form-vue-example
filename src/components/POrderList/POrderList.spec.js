import { shallowMount, createLocalVue } from '@vue/test-utils'

import POrderList from './POrderList'

const localVue = createLocalVue()

const mockedOrders = [
  {
    name: 'Suco de Goiaba',
    flavour: 'Goiaba',
    price: 5,
    description: 'Um mero suco de goiaba. Tem sachet de açúcar no balcão...',
    image: { preview: 'image' },
    type: 'drinks'
  }
]

describe('POrderList component', () => {
  let wrapper
  beforeAll(() => {
    wrapper = shallowMount(POrderList, {
      localVue
    })
  })

  it('should match the snapshot', () =>
    expect(wrapper.html()).toMatchSnapshot())

  it('should have an element with .order-list class', async () => {
    expect(wrapper.find('.order-list').exists()).toBe(true)
  })

  it('should render "POrderList" component when order prop is received', async () => {
    await wrapper.setProps({ orders: mockedOrders })
    expect(wrapper.findComponent({ name: 'POrderItem' }).exists()).toBe(true)
  })
})
