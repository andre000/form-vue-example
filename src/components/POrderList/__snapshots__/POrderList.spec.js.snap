// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POrderList component should match the snapshot 1`] = `<ul class="order-list"></ul>`;
